packages/core/core/src/services/session-manager.ts

```diff
@@ -197,13 +210,49 @@ class SessionManager {
         sessionId: payload.sessionId,
       };
     } catch (error: any) {
+      // If the token is expired, verify signature ignoring expiration to safely extract payload
+      // and clean up the corresponding session in database.
       if (error instanceof jwt.JsonWebTokenError) {
+        if (error.name === 'TokenExpiredError') {
+          try {
+            const expiredPayload = jwt.verify(token, this.config.jwtSecret, {
+              // Validate signature but ignore exp to retrieve session information
+              ignoreExpiration: true as any,
+            }) as RefreshTokenPayload;
```

Comment:
Instead of `as` can we add type checking in case `verify` returns a string? (even though we know it shouldn't with how we're using it)